### DISCUSSION TITLE
markRead가 작동하지 않습니다.

### DISCUSSION THREAD

**Discussion by h2saj5xz** [storycraft/node-kakao#239] [Q&A] [answered]
채널아이디로 채널객체를 만들고
해당 채널에 메시지를 보내기 전에
이전 글들을 전부 읽음처리 하고
메시지를 보내도록 코드를 짰는데,
읽음처리가 되질 않습니다.

const logId = await channel.info.lastChatLogId;
await channel.markRead(logId);

위 코드를 기본으로
중괄호로 감싸서도 보내보고
혹시나해서 롱타입 객체로 만들어서도 보내보고
보내기전에 채널 업데이트도 해보고
바로 직전 챗이 상대방인 상태로도 해봤는데
읽음처리가 되질 않습니다.

한번 읽음처리가 된적도도 있었는데
다음 챗을 보내니 다시 원래대로 돌아갔습니다.
어떻게 해야 할까요 ㅠㅠ

**Comment by ghost** [comment]
markRead는 ChatLogged를 매개변수로 넣어줘야 합니다.
ChatLogged는 { logId: Long } 으로 기억하고 있습니다.
그리고 확인해본 결과 logId를 받아오는데에는 await이 필요하지 않습니다.

```js
const logId = channel.info.lastChatLogId;
await channel.markRead({ logId });
```

으로 해주는게 맞지 않을까 싶습니다.

**Reply by h2saj5xz** [reply]
의견 감사합니다. 문서 보니 ChatLogged 객체로 보내야하는 것 같아서 중괄호로 감싸서 보내봤었습니다만 성공하지 못했었습니다.

**Comment by ghost** [comment, marked as answer]
모바일버전 카카오톡에선 syncChatList으로 읽음처리를 하는것 같습니다.
제 예상으로는 cur부터 max사이에 채팅을 읽음처리를 하는것 같습니다.

**Reply by ghost** [reply]
패킷을 보내본 결과 읽음처리됩니다.

**Reply by h2saj5xz** [reply]
아 제가 모바일 버전으로 쓰고 있었는데, 그래서 그런가보네요.
감사합니다!
